tools/release-plugin.sh

```diff
@@ -67,30 +57,68 @@ while getopts "v:abh" opt; do
 done
 shift "$(($OPTIND -1))"
 
-# If there are more than two arguments, bail.
-[[ $# -gt 2 ]] && die "Too many arguments specified! Only provide a project and a version number. Got:$(printf ' "%s"' "$@")"$'\n'"(note all options must come before the project slug)"
+# Parse arguments in associated array of plugins/project => version format.
+if [ $# -eq 0 ]; then
+	usage
+fi
+
+declare -A PROJECTS
+while [[ $# -gt 0 ]]; do
+	if [[ "$1" =~ ^[a-zA-Z\-]+$ ]]; then
+		PLUGIN=$1
+	else
+		die "Script must be passed in <project> [version] format. Got $1"
+	fi
+
+	if [[ "$2" =~ ^[0-9]+(\.[0-9]+)+(-.*)?$ ]]; then
+		PROJECTS["$PLUGIN"]=$2
+		SHIFT="2"
+	else
+		PROJECTS["$PLUGIN"]=''
+		SHIFT="1"
+	fi
+	shift "$SHIFT"
+done
 
-# Get the project slug.
-PROJECT=
-SLUG="${1#projects/}" # DWIM
-SLUG="${SLUG%/}" # Sanitize
-if [[ -e "$BASE/projects/$SLUG/composer.json" ]]; then
-	yellow "Project found: $SLUG"
-	PROJECT="$SLUG"
+# If we're releasing Jetpack, we're also releasing mu-wpcom-plugin.
+if [[ -v PROJECTS["jetpack"] && ! -v PROJECTS["mu-wpcom-plugin"] ]]; then
+	PROJECTS["mu-wpcom-plugin"]=''
 fi
-[[ -z "$PROJECT" ]] && die "A valid project slug must be specified (make sure project is in plugins/<project> format and comes before specifying version number."
 
-# Determine the project version
-[[ -z $2 ]] && die "Please specify a version number"
-# Figure out the version(s) to use for the plugin(s).
+# Check that the projects are valid.
+for PLUGIN in "${!PROJECTS[@]}"; do
+	# Get the project slug.
+	SLUG="${PLUGIN#projects/}" # DWIM
+	SLUG="${SLUG%/}" # Sanitize
+	SLUG="plugins/$SLUG"
+	if [[ ! -e "$BASE/projects/$SLUG/composer.json" ]]; then
+		die "$SLUG isn't a valid project!"
```

Comment:
@anomiex Maybe I'm a bit confused, but running with `tools/release-plugin.sh weekly` seems to exit here.

Reply:
You are a bit confused. You're supposed to do like `tools/release-plugin.sh jetpack mu-wpcom-plugin` and the script will decide that "weekly" is the prefix to use.

Or, for this specific case, `tools/release-plugin.sh jetpack` will work because it's coded to assume `mu-wpcom-plugin` always goes with Jetpack.